dds_web/security/project_user_keys.py

```diff
@@ -1,13 +1,43 @@
 """ Code for generating and maintaining project and user related keys """
 import os
 
+import argon2
 import cryptography.exceptions
 from cryptography.hazmat.primitives import asymmetric, ciphers, hashes, serialization
 import flask
 import gc
 
 from dds_web.database import models
-from dds_web.errors import KeyNotFoundError
+from dds_web.errors import (
+    KeyNotFoundError,
+    KeyOperationError,
+    KeySetupError,
+    SensitiveContentMissingError,
+)
+from dds_web.security.auth import (
+    extract_encrypted_token_sensitive_content,
+    extract_token_invite_key,
+)
+
+
+def __derive_key(user, password):
+    if user.kd_salt is None:
+        raise KeySetupError(message="User keys are not properly setup!")
+
+    derived_key = argon2.low_level.hash_secret_raw(
+        secret=password.encode(),
+        salt=user.kd_salt,
+        time_cost=2,
+        memory_cost=flask.current_app.config["ARGON_MEMORY_COST"],
```

Comment:
@talavis @inaod568 
May I rename `ARGON_MEMORY_COST` as `ARGON_KD_MEMORY_COST` where KD stands for key derivation.
Ina and I talked about tweaking the parameters of argon password hasher as it was below the recommendations. I think no issue has been created for this yet. I suggest we have two config variables one for key derivation and the other for password hasher which may be abbreviated as PH. Or if it doesn't bother you, we may just type it long with all the words.

Reply:
@worukan That's fine with me.

Reply:
Perfectly fine with me